### PR TITLE
feat(container): send POST to CI Sense for container remote run

### PR DESCRIPTION
**Pull request by MarkusZoppelt**
### Motivation/Context

The `/v3/runs` endpoint in CI Sense expects us to POST a container remote run.

### Description

* add API structs and func to communicate with the `/v3/runs` endpoint
* send request after container build during `cifuzz container remote-run`

### How to use/reproduce

Ensure you are logged in to CI Sense:

    cifuzz login --server app.staging.code-intelligence.com

build this branch and the corresponding container image:

    make install
    make build-container-image

Start a local container registry for testing:

    docker run -d -p 5000:5000 --name cifuzz-test-registry registry:2

Try container remote run and see the successfull POST in `verbose` mode:

    cifuzz container remote-run -C examples/maven --registry localhost:5000/test/cifuzz  com.example.FuzzTestCase::myFuzzTest --server app.staging.code-intelligence.com -v

### Tests

    make test/e2e

### CODE REVIEW

File: e2e/docker.go [hunk 1]
```diff
@@ -77,11 +77,11 @@ func buildImageFromDockerFile(t *testing.T, ctx context.Context, dockerClient *c
 }
 
 func runTestCaseInContainer(t *testing.T, ctx context.Context, dockerClient *client.Client, testCase *TestCase,
-	testCaseRun testCaseRunOptions, imageTag string) CommandOutput {
+	testCaseRun testCaseRunOptions, imageTag string, server string) CommandOutput {
 
 	t.Helper()
 	if testCase.CIUser != AnonymousCIUser {
-		testCaseRun.args = "--server=" + ciServerToUseForE2ETests + " " + testCaseRun.args
+		testCaseRun.args = "--server=" + server + " " + testCaseRun.args
```
Comment (jochil) [inline e2e/docker.go:84]: just a thought.. maybe using `fmt.Sprintf` here?

**Review by jochil — CHANGES_REQUESTED** [review]
Great instructions for reproducing it, thanks!
But I got the following error:
```
🔍 POST container remote run to https://app.staging.code-intelligence.com: {"image":"localhost:5000/test/cifuzz:b13844d9f886a97da753106e058e34ba41b2b7d3e0e47947a12d4b83ad0072b3","fuzz_tests":[{"name":"com.example.FuzzTestCase::myFuzzTest"}],"project_external_id":"Foo%20Bar-5ce01809"}
❌ 400 Bad Request: No such project
```
I tried other projects and depending on the project name the result changed. So every time there is a white space the project name I got a `400` response.
```
Foo Bar -> 400
Test123 -> 200
Android Example -> 400
Test1 -> 200
```
Can you check this please? Maybe the API expects the name in the json body to be not url encoded?

File: internal/api/container_run.go [hunk 2]
```diff
@@ -40,3 +50,38 @@ type Job struct {
...
+		FuzzTests:         tests,
+		ProjectExternalID: project,
+	}
+
+	body, err := json.Marshal(containerRun)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	log.Debugf("POST container remote run to %s: %s\n", client.Server, string(body))
```
[the remark this answers is not in the document]
Comment (MarkusZoppelt) [reply internal/api/container_run.go:70]: I like the idea. Should I leave it as is and wait for your incoming changes?

File: internal/api/container_run.go [hunk 3]
```diff
@@ -40,3 +50,38 @@ type Job struct {
...
+	for _, fuzzTest := range fuzzTests {
+		tests = append(tests, FuzzTest{Name: fuzzTest})
+	}
+	containerRun := &ContainerRun{
+		Image:             image,
+		FuzzTests:         tests,
+		ProjectExternalID: project,
+	}
+
+	body, err := json.Marshal(containerRun)
```
[the remark this answers is not in the document]
Comment (MarkusZoppelt) [reply internal/api/container_run.go:65]: Cool idea. I would like to keep the concise output for our `--plain` style output and your proposed solution for `--pretty` (default).

File: internal/api/container_run.go
[the remark this answers is not in the document]
Comment (MarkusZoppelt) [reply internal/api/container_run.go:1]: I will replace all custom `Types` with `*Types`

File: internal/cmd/container/remoterun/remoterun.go [hunk 4]
```diff
@@ -121,3 +195,28 @@ func (c *containerRemoteRunCmd) buildImage() (string, error) {
 
 	return container.BuildImageFromBundle(bundlePath)
 }
+
+func (c *containerRemoteRunCmd) selectProject(projects []*api.Project) (string, error) {
```
[the remark this answers is not in the document]
Comment (MarkusZoppelt) [reply internal/cmd/container/remoterun/remoterun.go]: I introduced two new functions in the `dialog` package that are reusable 👍🏼

File: internal/api/container_run.go
[the remark this answers is not in the document]
Comment (MarkusZoppelt) [reply internal/api/container_run.go:1]: Done

File: internal/api/container_run.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (MarkusZoppelt) [reply internal/api/container_run.go:63]: And maybe in another PR. This one is already changing a bunch of files in different places. Will create a ticket.

[the remark this answers is not in the document]
Comment (MarkusZoppelt) [reply internal/api/container_run.go:64]: CLI-1280

**Review by jochil — APPROVED** [review]
thanks

**Pull request merged**